2_adventures/001/lenn4rd/03_raindrops_exercise/raindrops.rb

```diff
@@ -0,0 +1,25 @@
+class Fixnum
+  def prime_factor_of?(number)
+    number % self == 0
+  end
+end
+
+class Raindrops
+  PRIME_FACTORS = [3, 5, 7]
+
+  SOUNDS_FOR_FACTORS = {
+    '3' => 'Pling',
+    '5' => 'Plang',
+    '7' => 'Plong'
+  }
+  
+  def self.convert(number)
+    PRIME_FACTORS.each_with_object(sound = '') do |factor|
+      sound << SOUNDS_FOR_FACTORS[factor.to_s] if factor.prime_factor_of?(number)
+    end
+    
+    return number.to_s if sound.empty?
+    
+    sound
```

Comment:
Actually I prefer `&&` and `||` over english words because they indicate that this line is a logic statement. I'll immediately recognize it, while using `and` and `or` would take me longer.

I don't know if it's true today or if I misunderstood it back then but in my early (earlier…) Ruby days I read somewhere that there's a difference between using `&&` and `and`, that those two expressions aren't identical. So I stopped using the words and just used the symbols as before just to be safe. It might have been something Rails-specific — I'll do some research on that topic.